frontend/packages/dev-console/src/components/catalog/WebTerminalConfiguration.tsx

```diff
@@ -0,0 +1,363 @@
+import { Dropdown } from '@console/internal/components/utils/dropdown';
+import { NumberSpinner } from '@console/internal/components/utils/number-spinner';
+import { DevWorkspaceTemplateModel } from '@console/internal/models';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { useTelemetry } from '@console/shared';
+import {
+  LoadError,
+  SaveStatus,
+  SaveStatusProps,
+} from '@console/shared/src/components/cluster-configuration';
+import { DEFAULT_NS_OPERATORS } from '../../const';
+
+const WebTerminalConfiguration = ({ readonly }) => {
+  const { t } = useTranslation();
+  const fireTelemetryEvent = useTelemetry();
+  const [operatorNamespace] = useCloudShellNamespace();
+  const [unit, setUnit] = React.useState<string>('m');
+  const [value, setValue] = React.useState<number>(0);
+  const [image, setImage] = React.useState('');
+  const [timeoutCheckBox, setTimeoutCheckBox] = React.useState(false);
+  const [imageCheckBox, setImageCheckBox] = React.useState(false);
+  const [saveStatus, setSaveStatus] = React.useState<SaveStatusProps>();
+
+  const webTerminalExecResource: WatchK8sResource = React.useMemo(() => {
+    return {
+      kind: referenceForModel(DevWorkspaceTemplateModel),
+      namespaced: true,
+      isList: false,
+      name: 'web-terminal-exec',
+      namespace: operatorNamespace || DEFAULT_NS_OPERATORS,
+    };
+  }, [operatorNamespace]);
+
+  const webTerminalToolingResource: WatchK8sResource = React.useMemo(() => {
+    return {
+      kind: referenceForModel(DevWorkspaceTemplateModel),
+      namespaced: true,
+      isList: false,
+      name: 'web-terminal-tooling',
+      namespace: operatorNamespace || DEFAULT_NS_OPERATORS,
+    };
+  }, [operatorNamespace]);
+
+  const [webTerminalExec, isLoaded, loadError] = useK8sWatchResource<K8sResourceKind>(
+    webTerminalExecResource,
+  );
+
+  const [
+    webTerminalTooling,
+    iswebTerminalToolingLoaded,
+    webTerminalToolingloadError,
+  ] = useK8sWatchResource<K8sResourceKind>(webTerminalToolingResource);
+
+  const terminalExecComponentIndex =
+    webTerminalExec &&
+    isLoaded &&
+    !loadError &&
+    webTerminalExec?.spec?.components?.findIndex((e) => e.name === 'web-terminal-exec');
+
+  const terminalToolingComponentIndex =
+    webTerminalTooling &&
+    iswebTerminalToolingLoaded &&
+    !webTerminalToolingloadError &&
+    webTerminalTooling?.spec?.components?.findIndex((e) => e.name === 'web-terminal-tooling');
+
+  React.useEffect(() => {
+    if (isLoaded && !loadError && terminalExecComponentIndex !== -1) {
+      const terminalExecComponent = webTerminalExec?.spec?.components[
+        terminalExecComponentIndex
+      ]?.container?.env?.findIndex((e) => e.name === 'WEB_TERMINAL_IDLE_TIMEOUT');
+      if (
+        terminalExecComponent !== -1 &&
+        webTerminalExec?.spec?.components[terminalExecComponentIndex]?.container?.env[
+          terminalExecComponent
+        ]?.value
```

Comment:
again I would expect that this will be easier if you use `find` instead of `findIndex`:

```suggestion
      const terminalExecComponent = webTerminalExec?.spec?.components[
        terminalExecComponentIndex
      ]?.container?.env?.find((e) => e.name === 'WEB_TERMINAL_IDLE_TIMEOUT');
      if (terminalExecComponent?.container?.env[terminalExecComponent]?.value
```